fog/ledger/enclave/impl/src/lib.rs

```diff
@@ -93,30 +136,154 @@ impl LedgerEnclave for SgxLedgerEnclave {
+        };
+
+        let rec2 = KeyImageData {
+            key_image: KeyImage::from(2),
+            block_index: 15867249514237159136,
+            timestamp: 14315610570481526166,
+        };
+
+        let rec3 = KeyImageData {
+            key_image: KeyImage::from(2),
+            block_index: 14978249314436157236,
+            timestamp: 14613610561491525175,
+        };
+
+        // record not added to oram
+        let not_found_rec = KeyImageData {
+            key_image: KeyImage::from(2),
+            block_index: 16967239515437158243,
+            timestamp: 13714610510481517185,
+        };
+
+        let key_image = &KeyImage::from(2); // create key image
+
+        // add test KeyImageData record to ledger oram
+        let v_result1: core::result::Result<_, AddRecordsError> =
+            keyimagestore.add_record(key_image, rec.block_index, rec.timestamp);
+
+        assert!(v_result1.is_ok() && !v_result1.is_err());
+        //create temp variables to store KeyImageData which we will use as key to query
+        // ledger oram with find_record
+
+        //query the ledger oram for the record using the key_image
+        let v = keyimagestore.find_record(key_image);
+
+        // this test should pass since we added this rec into the oram
+        assert_eq!(rec.block_index, v.spent_at);
+        assert_eq!(rec.timestamp, v.timestamp);
+        assert_eq!(
+            v.key_image_result_code,
+            fog_types::ledger::KeyImageResultCode::Spent as u32
+        );
+
+        // this test should pass since we did not add this rec into the oram
+        assert_ne!(not_found_rec.block_index, v.spent_at);
+        assert_ne!(not_found_rec.timestamp, v.timestamp);
+
+        let key_image2 = &KeyImage::from(2); // create key image
+
+        // add test KeyImageData record to ledger oram
+        let v_result2: core::result::Result<_, AddRecordsError> =
+            keyimagestore.add_record(key_image2, rec2.block_index, rec2.timestamp);
+
+        assert!(v_result2.is_ok() && !v_result2.is_err());
+
+        let key_image3 = &KeyImage::from(2); // create key image
+                                             // add test KeyImageData record to ledger oram
+        let v_result3: core::result::Result<_, AddRecordsError> =
+            keyimagestore.add_record(key_image3, rec3.block_index, rec3.timestamp);
+
+        assert!(v_result3.is_ok() && !v_result3.is_err());
+        //we can add the record even if the key image is all zero bytes
+        let rec3 = KeyImageData {
+            key_image: KeyImage::from(0),
+            block_index: 14978249314436157236,
+            timestamp: 14613610561491525175,
+        };
+
+        let v_result: core::result::Result<_, AddRecordsError> =
+            keyimagestore.add_record(&rec3.key_image, rec3.block_index, rec3.timestamp);
+
+        // we should not get back "invalid key" error
+        assert!(!v_result.is_err() && v_result.is_ok());
```

Comment:
it can fail if the ORAM overflows

the test could be improved if each `add_record` is also followed by up with a `find_record` call, and `find_record` for other keys.